Add Rocky Linux 9 support

lumarel opened the pull request:
## What does this PR change?

This PR should make Rocky Linux 9 work completely.

I hope I didn't miss something took me a while to get through all parts of the application.

The only thing I didn't add are the parts in the 2 `additional_products.json` files and the referencing repo templates in `mgr-bootstrap-data.py`, as I didn't know which IDs I would have had to set.
If you could tell me which I should set there I can of course also add this right away 🙂 

It also adds the EPEL 9 key to the schema as this hasn't been done up to now, as far as I can see.

## GUI diff

No difference.

## Documentation
- Documentation issue was created: [uyuni-docs#1661](https://github.com/uyuni-project/uyuni-docs/issues/1661)
- (OPTIONAL) [Documentation PR](https://github.com/uyuni-project/uyuni-docs/pulls)

- [ ] **DONE**

## Test coverage
- No tests: already covered

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

- [x] **DONE**

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

I don't really know which tests to run.

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

juliogonzalez commented on utils/spacewalk-common-channels.ini:
```diff
@@ -3567,3 +3567,110 @@ gpgkey_url = http://download.fedoraproject.org/pub/epel/RPM-GPG-KEY-EPEL-8
+base_channels = rockylinux9-%(arch)s
+repo_url = https://mirrors.rockylinux.org/mirrorlist?repo=RT-9&arch=%(arch)s
+
+[rockylinux9-saphana]
```
Does this actually exist? According to https://wiki.rockylinux.org/rocky/repo/, this is TBD, same for SAP and NFV

lumarel replied:
Yes it does, I at the one part looked at the URIs in the actual files on a running system, but also checked them before adding 👍🏻

lumarel commented on schema/spacewalk/upgrade/susemanager-schema-4.3.12-to-susemanager-schema-4.3.13/010-uyuni-gpg-key-url-update.sql:
```diff
@@ -98,3 +98,7 @@ gpg_key_url = 'https://repo.almalinux.org/almalinux/RPM-GPG-KEY-AlmaLinux';
 update rhnchannel set
 gpg_key_url = 'file:///etc/pki/rpm-gpg/RPM-GPG-KEY-rockyofficial' where
 gpg_key_url = 'https://dl.rockylinux.org/pub/rocky/RPM-GPG-KEY-rockyofficial';
+
```
Okay I will remove that later 👍🏻

lumarel replied:
I think that should be fixed now!

lumarel commented on utils/spacewalk-common-channels.ini:
```diff
@@ -3567,3 +3567,110 @@ gpgkey_url = http://download.fedoraproject.org/pub/epel/RPM-GPG-KEY-EPEL-8
+gpgkey_url = https://download.fedoraproject.org/pub/epel/RPM-GPG-KEY-EPEL-9
+gpgkey_id = 3228467C
+gpgkey_fingerprint = FF8A D134 4597 106E CE81 3B91 8A38 72BF 3228 467C
+repo_url = https://mirrors.fedoraproject.org/mirrorlist?repo=epel-9&arch=%(arch)s
```
Devel is basically every package from every repo + the packages which were needed for building everything, I don't think this belongs in a managed system, better to directly query it if needed 🙂 
And rockypi is as the name suggests the Raspberry Pi repo, which hm... yeah I could add this, if someone wants to integrate these systems (Raspberry Pi 2+ 64-bit edition, 3, 4) 👍🏻

lumarel replied:
Okay, RockyPi will be a SIG in the future, so we can add it later, but it's not present up to now

lumarel commented:
@juliogonzalez 

Sorry I didn't have the time to test that yesterday then, was quite late,
will change the few things this evening and run some tests, thank you for the guidance I wouldn't have known how to do them anyway 🙂

juliogonzalez commented:
> @juliogonzalez
> 
> Sorry I didn't have the time to test that yesterday then, was quite late, will change the few things this evening and run some tests, thank you for the guidance I wouldn't have known how to do them anyway slightly_smiling_face

One day I will write a quick dock about how to build the packages.

But for now it's easier if you find each of the file names at the server with `find`, and then replace (always remember to stop services first). That's what I do for quick testing.

lumarel commented:
I already know where most of the files are, due to me reverse engineering here and there (as I didn't understand why something was not working), so I think I will find the files quite easily only the compiled java stuff is a bit of a hard part for me 🙂

I unfortunately don't have a running Uyuni DEV system right now, and also failed to set one up today (only have RL8 dev systems), so tomorrow I will give it another one... hopefully no more hostname failures 🙂

lumarel commented:
@juliogonzalez I got a system running again! And tested the stuff, looks good, only was a bit finicky as the bootstrap and common-channels files look to have changed quite a bit during die month 🙂

lumarel commented:
Rebasing done 🙂

lumarel commented:
Another rebase, is there something more to do?

juliogonzalez reviewed (commented):
LGTM, but we'll merge it later, after 2022.07 is released. I will also ask you to change `100-rocky-epel.sql` to a new schema directory, as we'll have a new one.

As mentioned at Gitter, it's OK if you already start preparing the doc :-)

juliogonzalez commented on schema/spacewalk/upgrade/susemanager-schema-4.3.12-to-susemanager-schema-4.3.13/100-rocky-epel.sql:
```diff
@@ -0,0 +1,11 @@
+-- Rocky Linux 9
```
Please move the script to `susemanager-schema-4.4.0-to-susemanager-schema-4.4.1/100-rocky-epel.sql`.

I think the folder does not exist yet, so you can create it.

juliogonzalez reviewed (changes requested):
We finally released 2022.08, so now you can move the script for the DB, and we can merge the PR.

If you can fix the conflicts for the changelogs, that would be great.

Otherwise I will do it before merging.

lumarel commented:
@juliogonzalez Thank you for the notification!
I rebased onto master now and changed the *.changes files.

And as you asked in the beginning about the Raspberry Pi stuff, I also added the altarch repos now, so if someone wants to play around with raspbis now, that should be easier now 👀

lumarel commented:
@juliogonzalez Fixed the conflicts again

juliogonzalez reviewed (approved):
LGTM

juliogonzalez commented:
@lumarel I approved, but just added mc as reviewer for the Java part, just in case.

Do not worry about the changelogs (unless mc requests more fixes), I will take care of fixing the conflicts for them.

mcalmer reviewed (approved):
Looks good. The other RHEL special code is version independent. It seems that the logic we implemented for Rock 8 will also work for Rocky 9

lumarel commented:
Thank you two! 🙂
